### ISSUE TITLE
Current month events are missing

### ISSUE THREAD

**Issue by jeremywight** [codeforgreenville/hackgreenville-com#85]
What: Navigate to the Calendar of events and notice that the current month (presently Feb 2021) is completely blank, while December and March have events. We currently have an event which should show up in Feb (Code for Greenville Hack Night) that is missing.

Determined and solve the bug of why it's not displaying this months events.

Example
![Calendar 2021-02-02 at 6 53 32 PM](https://user-images.githubusercontent.com/8245600/106678083-fa581900-6587-11eb-8c2d-61ac68bf7452.jpg)

**Comment by allella** [comment]
@zach2825 I haven't looked into this, but if you think there's a caching / cron job issue, then please let me know.

Thanks

**Comment by jdonboch** [comment]
This seems to be a problem with the API as you can recreate it by just hitting the API directly:

For example, this is the query that is made when you view the February calender but it doesn't return things in February and actually returns things beyond the March 14th end parameter.  
`https://hackgreenville.com/api/calendar?start=2021-01-31T00%3A00%3A00-05%3A00&end=2021-03-14T00%3A00%3A00-05%3A00`

If you slide the start request parameter to a couple weeks earlier, it starts returning data for February.
`https://hackgreenville.com/api/calendar?start=2021-01-31T00%3A00%3A00-05%3A00&end=2021-03-14T00%3A00%3A00-05%3A00`

The API is not interpreting the parameter range correctly for some reason.

**Comment by allella** [comment]
@zach2825 I believe this would be caching related since this "API" is internal to the HG app. Thoughts?

**Comment by allella** [comment]
I ran the following and it didn't seem to fix anything on the missing events

```
git pull && composer install; php artisan test && php artisan migrate
php artisan pull:events
php artisan cache:clear

```

**Comment by jdonboch** [comment]
After reviewing the code, I'm fairly certain that this bug deals with [Lines 17 and 18 in the Calendar API controller](https://github.com/codeforgreenville/hackgreenville-com/blob/c360a8fe667b63057aa789f1aefc32c007b2bcfe/app/Http/Controllers/Api/CalendarController.php#L17) since the front end is passing a start and end parameter but the end parameter is completely ignored and the start has some weird math where a month is added.  There are many posts about Carbon's add/sub month being not intuitive if you add a month to date and the following month has less days.  This seems to be the case since the start date is 1/31 but February does not have a 31st day so adding a month to 1/31 results in a date of March 3rd or 4th which is the exact behavior we are seeing (only dates from after March 4th and after are displayed).

I believe the correct solution would be to add a new method to the `Event` class that that can accept a start and end date and then updating the `Api/CalendarController` to use that method for getting events when both the start and end dates are provided.

I think I can figure out how to make the changes but I have no clue how to run this locally and test the changes, are there any documentation on that?

**Comment by zach2825** [comment]
There seemed to be some random bug in Carbon where I had logic to add a month; for some reason, it added 2 months in some scenarios. I simplified that logic to use the dates from the calendar.

**Comment by allella** [comment]
Closed via #89
